src/main/java/com/googlecode/lanterna/terminal/swing/AWTTerminalFontConfiguration.java

```diff
@@ -70,61 +76,54 @@
-                new Font("Liberation Mono", Font.PLAIN, getFontSize()),
-                new Font("VL Gothic Regular", Font.PLAIN, getFontSize()),
-                new Font("NanumGothic", Font.PLAIN, getFontSize()),
-                new Font("WenQuanYi Zen Hei Mono", Font.PLAIN, getFontSize()),
-                new Font("WenQuanYi Zen Hei", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing TW", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing HK", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing CN", Font.PLAIN, getFontSize())));
+                new Font("Ubuntu Mono", Font.PLAIN, fontSize),
+                new Font("FreeMono", Font.PLAIN, fontSize),
+                new Font("Liberation Mono", Font.PLAIN, fontSize),
+                new Font("VL Gothic Regular", Font.PLAIN, fontSize),
+                new Font("NanumGothic", Font.PLAIN, fontSize),
+                new Font("WenQuanYi Zen Hei Mono", Font.PLAIN, fontSize),
+                new Font("WenQuanYi Zen Hei", Font.PLAIN, fontSize),
+                new Font("AR PL UMing TW", Font.PLAIN, fontSize),
+                new Font("AR PL UMing HK", Font.PLAIN, fontSize),
+                new Font("AR PL UMing CN", Font.PLAIN, fontSize)));
     }
 
     private static List<Font> getDefaultFonts() {
+        int fontSize = getFontSize();
         return Collections.unmodifiableList(Collections.singletonList(
-                new Font("Monospaced", Font.PLAIN, getFontSize())));
+                new Font("Monospaced", Font.PLAIN, fontSize)));
     }
 
     // Here we check the screen resolution on the primary monitor and make a guess at if it's high-DPI or not
-    private static Integer CHOSEN_FONT_SIZE = null;
-    private synchronized static int getFontSize() {
-        if(CHOSEN_FONT_SIZE != null) {
-            return CHOSEN_FONT_SIZE;
-        }
-        // Source: http://stackoverflow.com/questions/3680221/how-can-i-get-the-monitor-size-in-java
-        GraphicsEnvironment ge      = GraphicsEnvironment.getLocalGraphicsEnvironment();
-        GraphicsDevice[]    gs      = ge.getScreenDevices();
-
-        // Assume the first GraphicsDevice is the primary screen (this isn't always correct but what to do?)
-        // Warning, there could be printers coming back here according to JavaDoc! Hopefully Java is reasonable and
-        // passes them in after the real monitor(s).
-        if (gs.length > 0) {
-            int primaryMonitorWidth = gs[0].getDisplayMode().getWidth();
-
-            // If the width is wider than Full HD (1080p, or 1920x1080), then assume it's high-DPI
-            if (primaryMonitorWidth > 2000) {
-                CHOSEN_FONT_SIZE = 28;
+    private static int getFontSize() {
+        if (Toolkit.getDefaultToolkit().getScreenResolution() >= 110) {
+            // Rely on DPI if it is a high value.
+            return Toolkit.getDefaultToolkit().getScreenResolution() / 7 + 1;
+        } else {
```

Comment:
Not sure the `else` makes sense anymore...?
So if the DPI is lower than 110, we're still doing the previous heuristics based on screen resolution? I really don't remember how reliable the DPI detection in Java is, would it be returning something below 110 if it doesn't know and that's why we should keep the below logic?

Reply:
The default value is somewhere around 90/96 on most older systems afaik. I think it's safe to keep the heuristics for a while as high DPI support by some OSes is still a bit new.